cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -1713,18 +1713,20 @@ std::pair<std::unique_ptr<column>, std::vector<column_name_info>> json_column_to
   return {};
 }
 
-table_with_metadata host_parse_nested_json(host_span<SymbolT const> input,
+table_with_metadata host_parse_nested_json(device_span<SymbolT const> d_input,
                                            cudf::io::json_reader_options const& options,
                                            rmm::cuda_stream_view stream,
                                            rmm::mr::device_memory_resource* mr)
 {
   // Range of orchestrating/encapsulating function
   CUDF_FUNC_RANGE();
 
-  auto const new_line_delimited_json = options.is_enabled_lines();
+  std::vector<SymbolT> h_input;
+  h_input.resize(d_input.size());
+  CUDF_CUDA_TRY(cudaMemcpyAsync(
+    h_input.data(), d_input.data(), h_input.size(), cudaMemcpyDefault, stream.value()));
```

Comment:
For my education -- where it is safe to use `cudaMemcpyDefault` instead of `cudaMemcpyXToY` (where `X`, `Y` are `Host` or `Device`)?

The docs say:
> Direction of the transfer is inferred from the pointer values. Requires unified virtual addressing.
https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1g18fa99055ee694244a270e4d5101e95b

but I am not confident about whether this means "always safe" or "only sometimes safe." Is UVA always available? If so, should we adopt `cudaMemcpyDefault` everywhere? If there is a performance cost or downside of some other kind, should we never use `cudaMemcpyDefault` and always use a specific host/device mode?

Reply:
Great question :)
I asked around and it seems like `cudaMemcpyDefault` is preferable if your system has unified virtual addressing (and it seems like only 32bit Windows does not). There's no performance downside.